src/sys/socket/ffi.rs

```diff
@@ -37,7 +37,7 @@ pub struct cmsghdr {
     pub cmsg_len: type_of_cmsg_len,
     pub cmsg_level: c_int,
     pub cmsg_type: c_int,
-    pub cmsg_data: [type_of_cmsg_len; 0]
+    pub cmsg_data: [usize; 0]
```

Comment:
Instead of using a standard rust type here and thus affecting (though probably not breaking) other platforms, you should fix the type definition in line 14 using the appropriate type alias from libc.